tests/e2e/specs/api/CloneGitRepoAPI.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { OpenshiftClientExecutor } from '../../utils/OpenshiftClientExecutor';
+import { expect } from 'chai';
+import { ShellString } from 'shelljs';
+import { GitUtil } from '../../utils/vsc/GitUtil';
+import { TestConstants } from '../../constants/TestConstants';
+
+const gitRepository: string = 'https://github.com/crw-qe/web-nodejs-sample';
+
+suite(`Test cloning of repo "${gitRepository}" into empty workspace.`, async function (): Promise<void> {
+    // works only for root user
+    const namespace: string = 'admin-devspaces';
+    const workspaceName: string = 'empty-' + Math.floor(Math.random() * 1000);
+    const clonedProjectName: string = GitUtil.getProjectNameFromGitUrl(gitRepository);
+    let containerWorkDir: string = '';
+
+    const openshiftClientExecutor: OpenshiftClientExecutor = new OpenshiftClientExecutor(workspaceName, namespace);
+    const containerTerminal: OpenshiftClientExecutor.ContainerTerminal = new OpenshiftClientExecutor.ContainerTerminal(openshiftClientExecutor);
+
+    const emptyYaml: string =
+        'apiVersion: workspace.devfile.io/v1alpha2\n' +
+        'kind: DevWorkspace\n' +
+        'metadata:\n' +
+        `  name: ${workspaceName}\n` +
+        'spec:\n' +
+        '  started: true\n' +
+        '  template:\n' +
+        '    components:\n' +
+        '      - container:\n' +
+        `          image: ${TestConstants.TS_DW_CONTAINER_IMAGE}\n` +
+        '        name: universal-developer-image\n' +
+        '  contributions:\n' +
+        '    - name: che-code\n' +
+        `      uri: ${TestConstants.TS_DW_CONTRIBUTIONS_URI}\n` +
```

Comment:
`TS_DW_CONTRIBUTIONS_URI` constant name is too vague.

In the current context, more relevant name could be `TS_API_TEST_CHE_CODE_EDITOR_DEVFILE_URI`.

Would you also consider setting up default value = `https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-code/latest/devfile.yaml` in TestConstants.ts file?

Reply:
Contributions is plural, and can reference anything, and test implementation specific.

The goal is to have customizable _UDI image_ and _Che Code Editor devfile URI_ address doesn't matter where it is applied to.
Then, the test could be run against upstream, downstream etc.

Reply:
sorry for misleading, no need to set constant default value in TestConstants.ts - it's already in there